components/esp_http_client/esp_http_client.c

```diff
@@ -439,6 +442,21 @@ static esp_err_t esp_http_client_prepare(esp_http_client_handle_t client)
     return ESP_OK;
 }
 
+static esp_err_t set_host_header(esp_http_client_handle_t client)
+{
+    if (!client->connection_info.host)
+        return ESP_ERR_INVALID_STATE;
+    // Always append the port; we need 1 + 5 + 1 = 7 extra characters beyond the length:
+    size_t size = strlen(client->connection_info.host) + 7;
+    char *buf = (char *)malloc(size);
+    if (!buf)
```

Comment:
According to ESP-IDF style guide, if statement should be followed by {} braces. So, I recommend changing this to
```
    if (!buf) {
         return ESP_ERR_NO_MEM;
    }
```